packages/flutter_tools/lib/src/asset.dart

```diff
@@ -788,23 +806,59 @@ class LicenseCollector {
           packageNames = <String>[package.name];
           licenseText = rawLicense;
         }
-        packageLicenses.putIfAbsent(licenseText, () => <String>{})
-          .addAll(packageNames);
+        packageLicenses.putIfAbsent(licenseText, () => <String>{}).addAll(packageNames);
         allPackages.addAll(packageNames);
       }
     }
+
     final List<String> combinedLicensesList = packageLicenses.keys
       .map<String>((String license) {
         final List<String> packageNames = packageLicenses[license].toList()
           ..sort();
         return packageNames.join('\n') + '\n\n' + license;
       }).toList();
     combinedLicensesList.sort();
-    final String combinedLicenses = combinedLicensesList.join(licenseSeparator);
+
+    /// Append additional LICENSE files as specified in the pubspec.yaml.
```

Comment:
would it be worth moving all the license-specific code into a class? since this code is sprinkled throughout this really long function, updating it would seem tricky.

Reply:
This is in the class LicenseCollector, I could definitely move it to a new library though.

Reply:
moved to a new library